lib/create-entity-finder-api.js

```diff
@@ -28,6 +28,21 @@ function getEntityInDataset(entityByDomain, entityByRootDomain, originOrURL) {
   return undefined
 }
 
+function getProductInDataset(entityByDomain, entityByRootDomain, originOrURL) {
+  const entity = getEntityInDataset(entityByDomain, entityByRootDomain, originOrURL);
+  const products = entity && entity.products;
+  if (!products) return undefined;
+  for (const product of products) {
+    for (const pattern of product.urlPatterns) {
+      const regex = new RegExp(pattern);
+      if (regex.test(originOrURL)) {
+        return product;
```

Comment:
Should we consider the case where a URL matches with more than one product?  We could easily return a list of all matching products instead of just the first one.

Reply:
I think we should construct products such that only one matches. 

the point of `.products` is supposed to be a disambiguation of an entity. I'm not sure what it would mean for multiple to match :)

Reply:
Alright I will keep the code as is assuming max one product will ever match.